File: construct-k-palindrome-strings/construct-k-palindrome-strings.java
class Solution {
   public boolean canConstruct(String s, int k) {
        int[] map = new int[26];
        for (int i = 0; i < s.length(); i++)
            map[s.charAt(i)-'a']++;
        int odd = 0;
        for (int i = 0; i < 26;i++)
            if (map[i]%2 == 1)
                odd++;
        
        return s.length() >= k && odd <= k;
    }
}

File: 1502-construct-k-palindrome-strings/construct-k-palindrome-strings.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        if (k > s.size()) return false;
        int letterCount[26] = {0}, oddCount = 0;
        for (char c : s) {
            oddCount += (++letterCount[c - 'a'] & 1) ? 1 : -1;
        }return oddCount <= k;
    }
};